hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/AbstractMapQueryMessageTask.java

```diff
@@ -231,4 +246,26 @@ private void collectResultsFromMissingPartitions(BitSet finishedPartitions, Coll
             }
         }
     }
+
+    // for rolling-upgrade compatibility, can be deleted in 4.0
+    private Operation createQueryOperation(Query query, Version clusterVersion, MapServiceContext mapServiceContext) {
```

Comment:
cant we put this method in DefaultMapOperationProvider to get rid of some code duplication?